cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/ReadGrant.java

```diff
@@ -0,0 +1,120 @@
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  $Revision: 4 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.inventory.permissions;
+
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+import org.opencadc.gms.GroupURI;
+
+/**
+ * Holds read grant information about an artifact.
+ * 
+ * @author majorb
+ *
+ */
+public class ReadGrant extends Grant {
+
+    // Is artifact read access available to anonymous (all) users.
+    private boolean anonymousAccess = false;
+
+    /**
+     * Construct a grant for the given artifactURI.
+     * @param artifactURI The applicable targetURI.
+     */
+    public ReadGrant(URI artifactURI) {
+        super(artifactURI);
+    }
+    
+    /**
+     * Get the read group list.
+     * @return The read groups.
+     */
+    public List<GroupURI> getGroups() {
+        return groups;
+    }
+
+    /**
+     * Get date after which the Grant is considered expired and should be renewed.
+     * @return The grant expiry date.
+     */
+    public Date getExpiryDate() {
+        return expiryDate;
+    }
+
+    /**
+     * Check if artifact access is available to anonymous (all) users.
+     * @return true if the artifact has anonymous access, false otherwise.
+     */
+    public boolean isAnonymousAccess() {
+        return anonymousAccess;
+    }
```

Comment:
How does one set anonymous access to true?  I guess you need to add 
 an anonymousAccess param to the constructor if it's required.

Reply:
At some point when getting the read permissions would there be a point where you can say 'the artifact is anonymous access, set the flag true, and return' was my thought?

Reply:
Yes, but I mean there is no way to set it to true in the object, is there?

Reply:
Method added to set anonymous access.

Reply:
No setters in these model objects... you can just make anonymousAccess a public attribute and remove the getter and setter.

Reply:
Agree if it's mandatory it should be in the constructor, added.